PracticeGit/Program.c

```diff
@@ -1,13 +1,50 @@
-// ------------------------------------------------------------------------------------------------
-// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
-// Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------------------
+#include <string.h>
+#include <ctype.h>
+#include <limits.h>
+#include "Program.h"
 
-#include <stdio.h>
+int PalindromeChecker (const char* str) {
+   size_t left = 0;
+   size_t right = strlen (str) - 1;
+   while (left < right) {
+      if (!isalnum ((unsigned char)str[left])) left++;
+      else if (!isalnum ((unsigned char)str[right])) right--;
+      else {
+         if (tolower ((unsigned char)str[left]) != tolower ((unsigned char)str[right])) return 0;
+         left++;
+         right--;
+      }
+   }
+   if (str[0] == '-') return 0;
```

Comment:
Why should this return 0?

Reply:
if this finds inputs as negative numbers, it returns not a palindrome i.e., 0

Reply:
If it's a negative number remove the negative sign and reverse it also check for palindrome. It's not invalid.